tools/sdk-generation-pipeline/packages/sdk-generation-lib/src/lib/publishResult.ts

```diff
@@ -0,0 +1,206 @@
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Connection, createConnection } from 'typeorm';
+
+import { getGenerateAndBuildOutput } from '../types/taskInputAndOuputSchemaTypes/GenerateAndBuildOutput';
+import { AzureBlobClient } from '../utils/blob/AzureBlobClient';
+import { requireJsonc } from '../utils/requireJsonc';
+import { logger } from '../utils/logger';
+import { AzureSDKTaskName } from '../types/commonType';
+import { TaskResultEntity, TaskResult } from '../types/taskResult';
+import { TaskResultDaoImpl } from '../utils/db/taskResultDaoImpl';
+import { TaskResultDao } from '../utils/db/taskResultDao';
+import { CodeGeneration } from '../types/codeGeneration';
+import { CodeGenerationDaoImpl } from '../utils/db/codeGenerationDaoImpl';
+import { CodeGenerationDao } from '../utils/db/codeGenerationDao';
+import { GenerateAndBuildOutput } from '../types/taskInputAndOuputSchemaTypes/GenerateAndBuildOutput'
+
+export async function sendSdkTaskResultToDB(pipelineBuildId: string, dbConnection: Connection, taskResults: TaskResult[]) {
+    if (pipelineBuildId === undefined) {
+        throw new Error('pipelineBuildId is empty!');
+    }
+
+    const taskResultDao: TaskResultDao = new TaskResultDaoImpl(dbConnection);
+    for (const taskResult of taskResults) {
+        await taskResultDao.put(pipelineBuildId, taskResult);
+    }
+}
+
+export async function sendSdkGenerationToDB(
+    dbConnection: Connection,
+    pipelineBuildId: string,
+    sdkGenerationName: string,
+    service: string,
+    serviceType: string,
+    language: string,
+    swaggerRepo: string,
+    sdkRepo: string,
+    codegenRepo: string,
+    triggerType: string,
+    status: string,
+    owner?: string,
+    tag?: string,
+    swaggerPR?: string,
+    codePR?: string
+) {
+    if (
+        pipelineBuildId === undefined ||
+        sdkGenerationName === undefined ||
+        service === undefined ||
+        serviceType === undefined ||
+        language === undefined ||
+        swaggerRepo === undefined ||
+        sdkRepo === undefined ||
+        codegenRepo === undefined ||
+        triggerType === undefined ||
+        status === undefined
+    ) {
+        throw new Error('SdkGeneration data is empty!');
+    }
+
+    const codeGenerationDao: CodeGenerationDao = new CodeGenerationDaoImpl(dbConnection);
+    const cg: CodeGeneration = new CodeGeneration();
```

Comment:
can you use 

    const cg: CodeGeneration = {
        name:  sdkGenerationName
        .....
    };
```

Reply:
> can you use
> 
> ```
> const cg: CodeGeneration = {
>     name:  sdkGenerationName
>     .....
> };
> ```

cg is an Entity of typeorm, some columns like 'id' will be generated automatically. Can't assign it directly to a json object.